src/lib/cat_stories/index.svelte

```diff
@@ -49,45 +55,194 @@
     openOverlayIndex = -1;
   }
 
-  function openCatOverlay(index) {
+  function openCatOverlay(index: number) {
     openOverlayIndex = index;
   }
+
+  let lastScrollLeft: number | null = null;
+  let lastScrollTimestampMS = 0;
+  let snapTimeout = null;
+  let periodicTimeout = null;
+  let periodicScrollLeft: number | null = null;
+
+  function scrollIfNotMoving(scrollCB: () => void) {
+    const storiesContainer = document.querySelector('.stories-container');
+    if (
+      periodicScrollLeft !== null &&
+      storiesContainer !== null &&
+      storiesContainer.scrollLeft === periodicScrollLeft
+    ) {
+      clearTimeout(periodicTimeout);
+      scrollCB();
+    } else {
+      periodicTimeout = setTimeout(() => scrollIfNotMoving(scrollCB), 50);
+    }
+    periodicScrollLeft = storiesContainer.scrollLeft;
+  }
+
+  function getNodeTransform(node: HTMLElement, center: number) {
+    const rect = node.getBoundingClientRect();
+    const val = Math.pow(
+      Math.cos(((Math.PI / 2) * (rect.left + rect.width / 2 - center) * 0.4) / center),
+      center <= 250 ? 1 : 2
+    );
+    const clamp = Math.max(0.75, Math.min(1, val));
+    return clamp;
+  }
+
+  function scaleChildren(childNodes, center) {
+    childNodes.forEach((_childNode) => {
+      const childNode = _childNode as HTMLElement;
+      if (childNode.childNodes.length > 0 && (childNode.childNodes[0] as HTMLElement).style) {
+        (childNode.childNodes[0] as HTMLElement).style.transform = 'scale(1)';
+        childNode.offsetHeight;
+        const clamp = getNodeTransform(childNode, center);
+        (childNode.childNodes[0] as HTMLElement).style.transform = `scale(${clamp})`;
+      }
+    });
+  }
+
+  function parseTransform(transformStr: string) {
+    const regexMatch = transformStr.match(/scale\((.*)\)/);
+    if (!regexMatch || regexMatch.length < 1) {
+      return 1;
+    }
+    return parseInt(regexMatch[0]);
+  }
+
+  function getLargestChild(parent: HTMLElement) {
+    let largest: HTMLElement | null = null;
+    for (const child of parent.childNodes) {
+      if (
+        (child.childNodes?.length > 0 && (child.childNodes[0] as HTMLElement)).style &&
+        (largest === null ||
+          parseTransform((child.childNodes[0] as HTMLElement).style.transform) >
+            parseTransform(largest.childNodes[0].style.transform))
+      ) {
+        largest = child as HTMLElement;
+      }
+    }
+    return largest;
+  }
+
+  function extractNameAndIndexFromId(carouselItem: HTMLElement) {
+    return carouselItem.id.split('-');
+  }
+
+  function getIdFromNameAndIndex(name: string, index: number) {
+    return `${name}-${index}`;
+  }
+
+  function getNeighboringNode(parent: HTMLElement, child: HTMLElement, next: boolean) {
+    const [childName, childIndex] = extractNameAndIndexFromId(child);
+    const childIdToFind = getIdFromNameAndIndex(childName, (childIndex + (next ? 1 : 2)) % 3);
+    return document.getElementById(childIdToFind);
```

Comment:
Using DOM operations like `getElementById`, `querySelector`, `querySelectorAll`, etc. in Svelte is usually an antipattern since we have better ways of getting underlying DOM elements.

Instead of doing this, I'd create an array that stores refs with something like (untested):

* index.svelte:
```svelte
<script>
  let els = {};
</script>

{#each new Array(3) as _, index}
  {#each CAT_DATA as catData}
    <CarouselItem {catData} {index} {els} />
  {/each}
{/each}
```

* CarouselItem.svelte:
```svelte
<script>
  export let catData, index, els;
</script>

<div bind:this={els[`${catData.name}-${index}`]}>
  ...
</div>
```